nomad/container.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import time
+
+import pylxd
+
+from . import constants
+from .exceptions import ContainerOperationFailed
+from .network import EtcHosts
+from .network import find_free_ip
+from .network import get_ipv4_ip
+from .provision import prepare_debian
+from .provision import provision_with_ansible
+from .provision import set_static_ip_on_debian
+
+logger = logging.getLogger(__name__)
+
+
+class Container(object):
+    """ Represents a specific container that is managed by LXD-Nomad. """
+
+    def __init__(self, project_name, homedir, client, name=None, **options):
+        self.project_name = project_name
+        self.homedir = homedir
+        self.client = client
+        self.name = name
+        self.options = options
+
+    #####################
+    # CONTAINER ACTIONS #
+    #####################
+
+    def destroy(self):
+        """ Destroys the container. """
+        container = self._get_container(create=False)
+        if container is None:
+            logger.info("Container doesn't exist, nothing to destroy.")
+            return
+
+        # Halts the container...
+        self.halt()
+        # ... and destroy it!
+        logger.info('Destroying container "{name}"...'.format(name=self.name))
+        container.delete(wait=True)
+        logger.info('Container "{name}" destroyed!'.format(name=self.name))
+
+    def halt(self):
+        """ Stops the container. """
+        if self.is_stopped:
+            logger.info('The container is already stopped.')
+            return
+
+        # Removes configurations related to container's hostnames if applicable.
+        self._unsetup_hostnames()
+
+        logger.info('Stopping...')
+        try:
+            self._container.stop(timeout=30, force=False, wait=True)
+        except pylxd.exceptions.LXDAPIException:
+            logger.warn("Can't stop the container. Forcing...")
+            self._container.stop(force=True, wait=True)
+
+    def provision(self, barebone=None):
+        """ Provisions the container. """
+        if not self.is_running:
+            logger.error('The container is not running.')
+            return
+
+        if barebone is None:  # None == only if the container isn't provisioned.
+            barebone = not self.is_provisioned
+
+        if barebone:
+            logger.info('Doing bare bone setup on the machine...')
+            prepare_debian(self._container)
+
+        logger.info('Provisioning container "{name}"...'.format(name=self.name))
+        for provisioning_item in self.options.get('provisioning', []):
+            logger.info('Provisioning with {0}'.format(provisioning_item['type']))
```

Comment:
Not that it matters much, but I'm curious as to why you've added the explicit `0` in `{0}`.